build.sh

```diff
@@ -1,8 +1,16 @@
-#!/bin/bash
+#!/bin/sh
+# build.sh: build SAW
+# usage: ./build.sh
+set -e
+
 git submodule update --init
 
 function install() {
-  cp $(find dist-newstyle -type f -name $1 | sort -g | tail -1) bin/
+  # XXX: this should probably sort by timestamp (to get the file just
+  # built) rather than version. (Also, -g will not necessarily process
+  # versions correctly, since it's intended for floats, and it may not
+  # be as portable as we'd like.)
+  cp $(find dist-newstyle -type f -name $1 -print | sort -g | tail -1) bin/
```

Comment:
To be honest, I wonder if we shouldn't just use the [`cabal list-bin`](https://cabal.readthedocs.io/en/stable/cabal-commands.html#cabal-list-bin) command instead of stringing together shell commands to achieve the same thing. I imagine that `cabal list-bin`'s would behavior would be much more predictable and portable.

Reply:
Yeah, that seems like a good plan

Reply:
Done in 3e915e262e22e3f905cfb0485cfd2ef527ac1db6

Reply:
(Make that e4d9ee2)